### PR TITLE
fix(aws_s3 source): Use the decoder to calculate type defs

### PR DESCRIPTION
**Pull request by fuchsnj**
Decoder support was added to the s3 source (https://github.com/vectordotdev/vector/pull/17098), but this change was not reflected in the type definitions. This is now fixed.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
There's an unused import, but otherwise 👍 Thanks for fixing!

**Review by jszwedko — APPROVED** [review]
Thanks!

File: scripts/integration/aws/compose-temp-xsHfqy.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+services:
```
[the remark this answers is not in the document]
Comment (fuchsnj) [reply scripts/integration/aws/compose-temp-xsHfqy.yaml]: I added a `.gitignore` for these

File: scripts/integration/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+*/compose-temp*.yaml
```
Comment (jszwedko) [inline scripts/integration/.gitignore:1]: 🎉

**Review by jszwedko — APPROVED** [review]

**Pull request merged**